src/main/java/org/zalando/nakadi/controller/CursorsController.java

```diff
@@ -31,44 +39,83 @@
-                             final FeatureToggleService featureToggleService) {
+                             final FeatureToggleService featureToggleService,
+                             final SubscriptionDbRepository subscriptionRepository,
+                             final EventTypeRepository eventTypeRepository) {
         this.cursorsService = cursorsService;
         this.featureToggleService = featureToggleService;
+        this.subscriptionRepository = subscriptionRepository;
+        this.eventTypeRepository = eventTypeRepository;
     }
 
     @RequestMapping(path = "/subscriptions/{subscriptionId}/cursors", method = RequestMethod.GET)
     public ResponseEntity<?> getCursors(@PathVariable("subscriptionId") final String subscriptionId,
-                                        final NativeWebRequest request) {
+                                        final NativeWebRequest request,
+                                        final Client client) {
         if (!featureToggleService.isFeatureEnabled(HIGH_LEVEL_API)) {
             return new ResponseEntity<>(NOT_IMPLEMENTED);
         }
-
         try {
+            validateSubscriptionReadScopes(client, subscriptionId);
             final List<SubscriptionCursor> cursors = cursorsService.getSubscriptionCursors(subscriptionId);
-            return status(OK).body(cursors);
+            return status(OK).body(new ItemsWrapper<>(cursors));
         } catch (final NakadiException e) {
             return create(e.asProblem(), request);
         }
     }
 
-    @RequestMapping(value = "/subscriptions/{subscriptionId}/cursors", method = RequestMethod.PUT)
+    @RequestMapping(value = "/subscriptions/{subscriptionId}/cursors", method = RequestMethod.POST)
     public ResponseEntity<?> commitCursors(@PathVariable("subscriptionId") final String subscriptionId,
                                            @RequestBody final List<SubscriptionCursor> cursors,
-                                           final NativeWebRequest request) {
+                                           final NativeWebRequest request,
+                                           final Client client) {
         if (!featureToggleService.isFeatureEnabled(HIGH_LEVEL_API)) {
             return new ResponseEntity<>(NOT_IMPLEMENTED);
         }
 
         try {
-            final boolean allCommitted = cursorsService.commitCursors(subscriptionId, cursors);
-            return allCommitted ? ok().build() : noContent().build();
+            validateSubscriptionReadScopes(client, subscriptionId);
+            final Map<SubscriptionCursor, Boolean> result = cursorsService.commitCursors(subscriptionId, cursors);
+            final List<CursorCommitResult> items = result.entrySet().stream()
+                    .map(entry -> new CursorCommitResult(entry.getKey(), entry.getValue()))
+                    .collect(Collectors.toList());
+            return ok(new ItemsWrapper<>(items));
```

Comment:
According to API, it should be 200 only if at least one cursor wasn't successfully committed.
Otherwise it should be 204.